macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
 			layer.contentsScale = scale
 		}
 	}
-	
-	/// Tint color of the button content.
-	@objc override public var contentTintColor: NSColor? {
+
+	private var format: ButtonFormat {
 		get {
-			if style == .primaryFilled {
-				return .white
-			} else {
-				return primaryColor
-			}
+			return ButtonFormat(
+				size: self.size,
+				style: self.style,
+				accentColor: self.accentColor
+			)
 		}
 		set {
-			super.contentTintColor = newValue
-			needsDisplay = true
+			self.accentColor = newValue.accentColor
+			self.style = newValue.style  // depends on accentColor
+			self.size = newValue.size
 		}
 	}
 
-	/// The primary color of the button, AKA, the fill color in the primaryFilled style, and the outline in the primaryOutline style
-	@objc public var primaryColor: NSColor = defaultPrimaryColor {
-		didSet {
-			if primaryColor != oldValue {
-				needsDisplay = true
-			}
-		}
-	}
+	/// Foreground text/image color when button is not pressed or disabled.
+	@objc public var contentTintColorRest: NSColor?
 
-	/// The primary style of the button
-	@objc public var style: ButtonStyle = .primaryFilled {
-		didSet {
-			if style != oldValue {
-				if style == .primaryFilled {
-					contentTintColor = .white
-				} else {
-					contentTintColor = primaryColor
-				}
-				needsDisplay = true
-			}
-		}
-	}
+	/// Foreground text/image color when button is pressed.
+	@objc public var contentTintColorPressed: NSColor?
```

Comment:
not necessary, but maybe we could incorporate the state-> color mapping into a dictionary that can be modified? Not sure if that's cleaner or works with objc though

Reply:
If the cleanliness factor is related to the accumulation of lots of properties like this one, what I would really like is for the color set to be a struct (possibly of structs, whatever) that replaces the `style` enum.  That would make it simpler for users to make and apply their own styles.  There are a lot of additional format parameters (padding, corner radius and more) that the designers eventually want to expose, and it would be great to aggregate them together somehow so that they can be applied easily on init or by setting a single property later.

Any suggestions on a way to do this that would work with ObjC?

Coming from python I know the pain of trying to do all this with dictionaries (or project plists)... one spends a lot of lines of code validating the data structure before one can do anything useful with it.

Reply:
dictionaries in swift (or objC with generics) are easier than python because you do have type guarantees. in general the way Apple has done this is to define an enum for the state and expose setters that take in an enum. Then the complication of whether a data structure or dictionary or whatever is needed to back it is just an implementation detail that doesn't leak into the API surface

Reply:
I did something like this for size mappings.